### PR TITLE
SONARJAVA-1434 RSPEC-3417 Disallowed dependencies should not be used

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyVersionMatcher.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        Integer local = versions[i];
+        Integer target = v.versions[i];
+
+        if (target == null) {
+          return true;
+        } else if (local == null) {
+          return false;
+        }
+
+        return (local > target) || (local == target && isAfterOrEqual(v, i+1));
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyVersionMatcher.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Add a new line at the end of this file.](https://github.com/SonarSource/sonar-java/blob/8a6c4ee4bd009d8e74a7db1a39d2f42c281898b0/java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyNameMatcher.java) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00113)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Add a new line at the end of this file.](https://github.com/SonarSource/sonar-java/blob/8a6c4ee4bd009d8e74a7db1a39d2f42c281898b0/java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyNameMatcher.java) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00113)

**Comment by m-g-sonar** [comment]
Postponed to 3.10

File: java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyAbstractMatcher.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      return false;
+    }
+    if (pattern == null) {
+      return true;
+    }
+    return pattern.matcher(attribute.getValue()).matches();
+  }
+
+  public static MavenDependencyAbstractMatcher alwaysMatchingMatcher() {
+    return new MavenDependencyAbstractMatcher() {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyAbstractMatcher.java]: MAJOR Make this a named "static" inner class. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request closed without merging**